### PR TITLE
Section6updates

### PR DESCRIPTION
**Pull request by JohanBester**
Here is a request to pull the Section 6 Updates into the development branch.

Please carefully review this work. A great deal many things were changed and there are probably issues that will need to be addressed and corrected.

Thank you.

### CODE REVIEW

File: Section06/6.1-SPAIntro.md [hunk 1]
```diff
@@ -49,37 +49,41 @@ While we are learning SPAs in "_plain_"/vanilla JS, starting with vanilla JS now
...
+![Basic SPA File Structure](Resources/img/basic-SPA-file-structure.png)
 
 <br>
 
 ### **Essential File Structure of an SPA**
 
 - **`components`** stands for "functional **components**." The `components` folder holds _JS modules_ which _export_ **functions** that populate the HTML document. Each functional component corresponds to a _section_ or page/_view_ of your website.
   - If your site has many "pages"/views, it is common to create a **`views` folder** inside `components` that contains the files for every view/"page".
-- **`lib`** represents the an _optional_ **lib**rary of _JS modules_ which _export_ Functions, Classes, Objects, or Arrays that we might need in any of our "main" (`components`, `store`, `index.js`) JavaScript files.
+- **`lib`** represents an _optional_ **lib**rary of _JS modules_ which _export_ Functions, Classes, Objects, or Arrays that we might maybe possibly want to use in any of our "main" (`components`, `store`, `index.js`) JavaScript files.
```
Comment (matsinet) [inline Section06/6.1-SPAIntro.md:82]: I would prefer if the `lib` folder was completely removed

File: Section06/6.1-SPAIntro.md [hunk 2]
```diff
@@ -98,30 +102,31 @@ As you build out your SPA you will have separate files for each view (or page th
 
 As you know, any JavaScript file gives a website functionality. The `index.js` file sets itself apart in what it gives functionality to. In the previously mentioned root `index.html`, there is a script tag in the body for the `index.js`. The reason for this is that the root `index.js` gives functionality for the entire web application that you create.
 
-![Basic SPA File Structure](img/basic-SPA-file-structure.png)
+![Basic SPA File Structure](Resources/img/basic-SPA-file-structure.png)
```
Comment (matsinet) [inline Section06/6.1-SPAIntro.md:105]: did we move the images for the markdown into the Resources folder?   Is that consistent over the entire curriculum?

File: Section06/6.2-SPAComponents.md [hunk 3]
```diff
@@ -192,16 +195,16 @@ To use the image imported, you reference the variable in the src attribute of th
...
-
-    export default () => html`
-        <section id="jumbotron">
-            <h2>Savvy Coders Jan. 2020 Cohort</h2>
-            <a href="">"Call to Action" "Button"</a>
-        </section>
-        <img class="cover-img" src="${sharksImage}" />
-    `;
+import html from "html-literal";
+import sharksImage from "/assets/IMG_20150708_133155453.jpg";
```
Comment (matsinet) [inline Section06/6.2-SPAComponents.md:199]: Were the quotes changed from single to double by a linter or by hand?   If by hand, why?

File: Section06/Activities/6.2-Activities/index.js [hunk 4]
```diff
@@ -3,68 +3,10 @@ document.querySelector(".fa-bars").addEventListener("click", () => {
...
-  img.alt = pic.title;
-  gallerySection.appendChild(img);
-});
-
 // handle form submission
-document.querySelector("form").addEventListener("submit", event => {
+document.querySelector("form").addEventListener("submit", (event) => {
   event.preventDefault();
-  Array.from(event.target.elements).forEach(el => {
+  Array.from(event.target.elements).forEach((el) => {
```
Comment (matsinet) [inline Section06/Activities/6.2-Activities/index.js:9]: were the parens added manually or via linter?   My linter suggests that single parameters do not have parens?

File: Section06/Activities/finalRootIndex.js [hunk 5]
```diff
@@ -0,0 +1,33 @@
...
+
+function render() {
+  document.querySelector("#root").innerHTML = `
+    ${Header()}
+    ${Nav()}
+    ${Main()}
+    ${Footer()}
+  `;
+
+  addEventListeners();
```
Comment (matsinet) [inline Section06/Activities/finalRootIndex.js:11]: I would like to rename the addEventListeners function to afterRender in all curriculum

**Review by matsinet — CHANGES_REQUESTED** [review]

File: Section06/6.1-SPAIntro.md (hunk 1 above)
Comment (JohanBester) [reply Section06/6.1-SPAIntro.md:82]: OK, will do!  :)

File: Section06/6.2-SPAComponents.md (hunk 3 above)
Comment (JohanBester) [reply Section06/6.2-SPAComponents.md:199]: Darn, I'm going to kill al the linters..

File: Section06/6.1-SPAIntro.md (hunk 2 above)
Comment (JohanBester) [reply Section06/6.1-SPAIntro.md]: The images have been moved to the correct space

File: Section06/Activities/6.2-Activities/index.js (hunk 4 above)
Comment (JohanBester) [reply Section06/Activities/6.2-Activities/index.js]: This has been corrected directly in GitHub

**Review by cbrantley4 — APPROVED** [review]
Looks good to me!

**Review by matsinet — APPROVED** [review]
Merge it

**Pull request merged**